examples/rendercontrol_opengl/window_singlethreaded.h

```diff
@@ -0,0 +1,127 @@
+**     from this software without specific prior written permission.
+**
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+** "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+** LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+** A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+** OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+** LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+** DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+** THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+** (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#pragma once
+
+#include <QtGui/QWindow>
+#include <QtCore/QTimer>
+
+class QOpenGLBuffer;
+class QOpenGLContext;
+class QOpenGLShaderProgram;
+class QOpenGLTexture;
+class QOpenGLVertexArrayObject;
+class QOffscreenSurface;
+class QQuickRenderControl;
+class QQuickWindow;
+class QQmlEngine;
+class QQmlComponent;
+class QQuickItem;
+
+class WindowSingleThreaded : public QWindow
+{
+    Q_OBJECT
+    Q_PROPERTY(qint64 titleBarHeight READ getTitleBarHeight CONSTANT)
+
+public:
+    WindowSingleThreaded();
+    ~WindowSingleThreaded();
+
+    Q_INVOKABLE
+    void setHitTestVisible(QQuickItem *item, bool value);
+    int getTitleBarHeight() const;
+
+protected:
+    void exposeEvent(QExposeEvent *e) override;
+    void resizeEvent(QResizeEvent *e) override;
+    void showEvent(QShowEvent *e) override;
+    void mouseMoveEvent(QMouseEvent *e) override;
+    void mousePressEvent(QMouseEvent *e) override;
+    void mouseReleaseEvent(QMouseEvent *e) override;
+    void keyPressEvent(QKeyEvent *e) override;
+    void keyReleaseEvent(QKeyEvent *e) override;
+
+private Q_SLOTS:
+    void run();
+
+    void createTexture();
+    void destroyTexture();
+    void render();
+    void renderGL();
+    void init();
+    void setupVertexAttribs();
+    void requestUpdate();
+    void handleScreenChange();
+
+private:
+    void startQuick(const QString &filename);
+    void updateSizes();
+    void resizeTexture();
+
+    QOpenGLContext *m_glContext{nullptr};
+    QOpenGLContext *m_sharedGLContext{nullptr};
+    QOffscreenSurface *m_offscreenSurface{nullptr};
+    QQuickRenderControl *m_renderControl{nullptr};
+    QQuickWindow *m_quickWindow{nullptr};
+    QQmlEngine *m_qmlEngine{nullptr};
+    QQmlComponent *m_qmlComponent{nullptr};
+    QQuickItem *m_rootItem{nullptr};
+    uint m_textureId{0};
+    QSize m_textureSize;
+    bool m_quickReady{false};
+    QTimer m_updateTimer;
+    qreal m_dpr{0};
+    bool m_isFWMInitalised{false};
+    int m_titleBarHeight{26};
```

Comment:
This value doesn't seem to be correct. 30 is a more appropriate value, I think.

Reply:
If you get this _26_ from my code, well, it refers to the caption height, not the whole title bar's height. the title bar height should be 30/31 on Windows.

Reply:
Changed to `30` in 7db1509021a5159dc14699bd60235aa51b67749c and it looks ok on my end.
![image](https://user-images.githubusercontent.com/29707457/168469441-bb70e8ea-bdd7-4e09-9ce8-28b0c4ef1816.png)